src/main/java/org/opensearch/sdk/ExtensionRouteHandlerFactory.java

```diff
@@ -0,0 +1,30 @@
+package org.opensearch.sdk;
+
+import org.opensearch.OpenSearchException;
+
+public class ExtensionRouteHandlerFactory {
+    private static ExtensionRouteHandlerFactory INSTANCE;
```

Comment:
Do you think we can use Guice here to achieve this?

Reply:
That would be great. I'll look into it. That main problem I was facing was that I needed to have the shortName of the extension available in the REST Handler (i.e. inside of RestHelloAction for the sample extension). The way I went about doing that was to create a singleton class that could be fetched throughout the SDK but if there is a better way to do it then I will switch to using that.

Reply:
>  if there is a better way to do it then I will switch to using that.

Extensions are always initialized with their settings. `getExtensionSettings()` must be implemented on the `Extension` interface.  (The `BaseExtension` class offers the convenience of either a settings object or yml file.)

Any method on the main`Extension` class ought to be able to call `getExtensionSettings().getShortExtensionName()` to access this value without any factory/singletons.

In the AD Extension we usually pass the `ExtensionsRunner` class to every REST handler, and we have a getter on the runner for the extension.  So all the AD REST handlers can do `runner.getExtension().getExtensionSettings().getShortExtensionName()`.